src/templates/infra/api/rest/index.ejs

```diff
@@ -1,27 +1,55 @@
-const { json } = require('body-parser')
-const { generateRoutes } = require('@herbsjs/herbs2rest')
 const cors = require('cors')
 const express = require('express')
-const renderShelfHTML = require('@herbsjs/herbsshelf')
-const usecases = require('../../../domain/usecases')
+const { json } = require('body-parser')
+const camelCase = require('lodash.camelcase')
 const db = require('../../../infra/data/database')
+const usecases = require('../../../domain/usecases')
+const renderShelfHTML = require('@herbsjs/herbsshelf')
+const { generateRoutes } = require('@herbsjs/herbs2rest')
 const repositoriesFactory = require('../../../infra/data/repositories')
 
-function prepareUsecases(usecases){
+function cloneUsecases(usecases){
     return Promise.all(usecases.map(uc => {
         const clonedUC = { ...uc}
         clonedUC.usecase = clonedUC.usecase({})()
         return clonedUC
     }))
 }
 
+const newRoute = (usecase, useId) => { 
+    const route = { usecase }
+    if(useId) route.id = 'id'
+    return route
+}
+
 async function prepareRoutes(config){
     const conn = await db.factory(config) 
     const repositories = await repositoriesFactory(conn)
 
-    const routes = <%- props.routes %>
+    // groupBy group
+    const ucByGroup = usecases.reduce((acc, obj) => {
+        if (Object.keys(acc).includes(obj.tags.group)) return acc
+        acc[obj.tags.group] = usecases.filter(x => x.tags.group === obj.tags.group)
+        return acc
+    }, {})
+
+    const routes = Object.keys(ucByGroup).map(group => {
+        const route = {
+            name: `'${camelCase(group)}'`
+        }
+        for (const obj of ucByGroup[group]) {
+            const uc = obj.usecase(repositories)
+            const ucDescription = uc().description.toLowerCase()
+
+            if (ucDescription.includes('create')) route.post = newRoute(uc)
+            else if (ucDescription.includes('update')) route.put = newRoute(uc)
+            else if (ucDescription.includes('delete')) route.delete = newRoute(uc)
+            else if (ucDescription.includes('find')) route.getById = newRoute(uc, true)
```

Comment:
how do you deal with `getAll`? 

https://herbsjs.org/docs/glues/herbs2rest#controller-list

Reply:
I dont implemented it because we dont have an getAll query into graphql layer, so I think it must to be implemented into another PR, and PR about getAll route/query

I just forgot to open a issue before, so I opened one right now https://github.com/herbsjs/herbs-cli/issues/21